### PR TITLE
NetworkCloud Azure-cli-extension 2024-07-01 API

### PR DESCRIPTION
**Pull request by priyamshet**
---
NetworkCloud Azure-cli-extension update for API version 2024-07-01 API 

Corresponding AAZ PR :https://github.com/Azure/aaz/pull/574 
This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [x] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [ ] Have you run `python scripts/ci/test_index.py -q` locally? (`pip install wheel==0.30.0` required)
- [x] My extension version conforms to the [Extension version schema](https://github.com/Azure/azure-cli/blob/release/doc/extensions/versioning_guidelines.md)

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your pull request is merged into main branch, a new pull request will be created to update `src/index.json` automatically.  
You only need to update the version information in file setup.py and historical information in file HISTORY.rst in your PR but do not modify `src/index.json`.

### CODE REVIEW

**Comment by yonzhan** [comment]
Thank you for your contribution! We will review the pull request and get back to you soon.

**Comment by mbashtovaya** [comment]
> Hi @priyamshet
> 
> ## ⚠️ Release Requirements
> ### Module: networkcloud
> * ⚠️ Remove `azext.isPreview: true` in azext_metadata.json for networkcloud
> 
> #### Notes
> * For more info about extension versioning, please refer to [Extension version schema](https://github.com/Azure/azure-cli/blob/release/doc/extensions/versioning_guidelines.md)

according to the mentioned document. 2.0.0b4 is the next prevuew version and should be marked with the isPreview=true. We need clarification as to why this is being flagged.

File: src/networkcloud/azext_networkcloud/azext_metadata.json [hunk 1]
```diff
@@ -1,4 +1,4 @@
 {
-    "azext.minCliCoreVersion": "2.61.0",
-    "azext.isPreview": true
+    "azext.isPreview": true,
+    "azext.minCliCoreVersion": "2.61.0"
```
Comment (AllyW) [inline src/networkcloud/azext_networkcloud/azext_metadata.json:3]: no change here actually?

**Review by kairu-ms — APPROVED** [review]

File: src/networkcloud/azext_networkcloud/azext_metadata.json (hunk 1 above)
Comment (priyamshet) [reply src/networkcloud/azext_networkcloud/azext_metadata.json:3]: There should be no change in that.

**Comment by mbashtovaya** [comment]
@necusjz can you help with merging?

**Pull request merged**